### PR TITLE
[dynamo, 3.12] enable dynamo-wrapped tests in CI

### PR DESCRIPTION
**Pull request by williamwen42**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #123307
* #124124
* #124100
* #124095

### CODE REVIEW

**Review by jansel — APPROVED** [review]

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -215,6 +215,11 @@ jobs:
           { config: "default", shard: 1, num_shards: 3, runner: "linux.2xlarge" },
           { config: "default", shard: 2, num_shards: 3, runner: "linux.2xlarge" },
           { config: "default", shard: 3, num_shards: 3, runner: "linux.2xlarge" },
+          { config: "crossref", shard: 1, num_shards: 2, runner: "linux.2xlarge" },
+          { config: "crossref", shard: 2, num_shards: 2, runner: "linux.2xlarge" },
```
Comment (malfet) [inline .github/workflows/pull.yml:219]: Do we really need crossref tests on 3.12 vs 3.11?

**Review by malfet — APPROVED** [review]
LGTM if CI is green. But please file a tasks to disable testing eventually on some older Python version. Also do we need to run crossref tests on 3.12?

File: .github/workflows/pull.yml (hunk 1 above)
Comment (williamwen42) [reply .github/workflows/pull.yml]: Removed for now

**Pull request closed without merging**